keyboards/bastardkb/charybdis/charybdis.c

```diff
@@ -0,0 +1,243 @@
+#endif
+            {
+                is_drag_scroll ^= 1;
+            }
+#ifndef CHARYBDIS_DRAGSCROLL_VARIABLE
+            pmw_set_cpi(is_drag_scroll ? CHARYBDIS_DRAGSCROLL_DPI : dpi_array[keyboard_config.dpi_config]);
+#else
+            pmw_set_cpi(is_drag_scroll ? (dpi_array[keyboard_config.dpi_config] * CHARYBDIS_DRAGSCROLL_MULTIPLIER) : dpi_array[keyboard_config.dpi_config]);
+#endif
+        break;
+    }
+
+/* If Mousekeys is disabled, then use handle the mouse button
+ * keycodes.  This makes things simpler, and allows usage of
+ * the keycodes in a consistent manner.  But only do this if
+ * Mousekeys is not enable, so it's not handled twice.
+ */
+#ifndef MOUSEKEY_ENABLE
+    if (IS_MOUSEKEY_BUTTON(keycode)) {
+        report_mouse_t currentReport = pointing_device_get_report();
+        if (record->event.pressed) {
+            currentReport.buttons |= 1 << (keycode - KC_MS_BTN1);
+        } else {
+            currentReport.buttons &= ~(1 << (keycode - KC_MS_BTN1));
+        }
+        pointing_device_set_report(currentReport);
+        pointing_device_send();
+    }
+#endif
+
+    return true;
+}
+
+// Hardware Setup
+void keyboard_pre_init_kb(void) {
+    scroll_inertia.x = 0;
+    scroll_inertia.y = 0;
+    
+    keyboard_pre_init_user();
+}
+
+void pointing_device_init(void) {
+    pmw_spi_init();
+    pmw_set_cpi(dpi_array[keyboard_config.dpi_config]);
+}
+
+void pointing_device_task(void) {
+    report_mouse_t mouse_report = pointing_device_get_report();
+    process_mouse(&mouse_report);
+    
+    if (is_drag_scroll) { 
+#ifdef CHARYBDIS_DRAGSCROLL_INVERT_X
+        scroll_inertia.x -= mouse_report.x;
+#else
+        scroll_inertia.x += mouse_report.x;
+#endif
+#ifdef CHARYBDIS_DRAGSCROLL_INVERT_Y
+        // Invert vertical scroll direction
+        scroll_inertia.y -= mouse_report.y;
+#else
+        scroll_inertia.y += mouse_report.y;
+#endif
+        // we only want to trigger scrolling once in 12 scrolls, for slower scroll
+        if(scroll_inertia.y > CHARYBDIS_TRACKBALL_SPEED_DIVIDER_Y || scroll_inertia.y < -CHARYBDIS_TRACKBALL_SPEED_DIVIDER_Y){
+            mouse_report.v = constrain(scroll_inertia.y / CHARYBDIS_TRACKBALL_SPEED_DIVIDER_Y * 2, 
+                                -CHARYBDIS_TRACKBALL_SPEED_DIVIDER_Y/2, CHARYBDIS_TRACKBALL_SPEED_DIVIDER_Y/2);
+            scroll_inertia.y = 0;
+        }
+        if(scroll_inertia.x > CHARYBDIS_TRACKBALL_SPEED_DIVIDER_X || scroll_inertia.x < -CHARYBDIS_TRACKBALL_SPEED_DIVIDER_X){
+            mouse_report.h = constrain(scroll_inertia.x / CHARYBDIS_TRACKBALL_SPEED_DIVIDER_X * 2, 
+                                -CHARYBDIS_TRACKBALL_SPEED_DIVIDER_X/2, CHARYBDIS_TRACKBALL_SPEED_DIVIDER_X/2);
+            scroll_inertia.x = 0;
+        }
+
+        mouse_report.x = 0;
+        mouse_report.y = 0;
+    }
+
+    pointing_device_set_report(mouse_report);
+    pointing_device_send();
+}
+
+void eeconfig_init_kb(void) {
+    keyboard_config.dpi_config = CHARYBDIS_DPI_DEFAULT;
```

Comment:
It may be worth doing this: 
```suggestion
	keyboard_config.raw = 0;
    keyboard_config.dpi_config = CHARYBDIS_DPI_DEFAULT;
```
That would zero out any values, and then the defaults can be set.